src/compound.jl

```diff
@@ -0,0 +1,59 @@
+struct CompoundSpecies end
+Symbolics.option_to_metadata_type(::Val{:iscompound}) = CompoundSpecies
+
+struct CompoundComponents end
+Symbolics.option_to_metadata_type(::Val{:components}) = CompoundComponents
+
+
+macro compound(species_expr, arr_expr...)
+    # Ensure the species name is a valid expression
+    if !(species_expr isa Expr && species_expr.head == :call)
+        error("Invalid species name in @compound macro")
+    end
+
+    # Parse the species name to extract the species name and argument
+    species_name = species_expr.args[1]
+    species_arg = species_expr.args[2]
+
+    # Construct the expressions that define the species
+    species_expr = Expr(:macrocall, Symbol("@species"), LineNumberNode(0), Expr(:call, species_name, species_arg))
+
+    # Construct the expression to set the iscompound metadata
+    setmetadata_expr = :($(species_name) = ModelingToolkit.setmetadata($(species_name), CompoundSpecies, true))
+
+    # Ensure the expressions are evaluated in the correct scope by escaping them
+    escaped_species_expr = esc(species_expr)
+    escaped_setmetadata_expr = esc(setmetadata_expr)
+
+    # Construct the array from the remaining arguments
+    arr = Expr(:vect,(arr_expr)...)
```

Comment:
Can you store the component information in a more easy to query manner. You could either have two metadata, `CompoundComponent` and `CompoundCoefficients` or have `CompoundComponent` return a vector of pairs of the form `[A => 3, B => 1, C => 2]` for A3BC2?

Reply:
@TorkelE and I spoke, and I think we prefer the approach of storing the symbolic components and their coefficients separately.

Reply:
Okay, will implement this in the next commit.